build.zig

```diff
@@ -1,60 +1,70 @@
 const std = @import("std");
 
-pub fn build(b: *std.Build) void {
-    const target = b.standardTargetOptions(.{});
+pub fn build(b: *std.Build) !void {
+    const target_query: std.Target.Query = .{};
+    const target = b.resolveTargetQuery(target_query);
     const optimize = b.standardOptimizeOption(.{});
 
-    _ = b.addModule("llvm", .{
-        .source_file = .{
-            .path = "src/llvm.zig",
-        },
+    const lib = b.addStaticLibrary(.{
+        .name = "llvm", 
+        .root_source_file = .{.path = "src/llvm.zig"},
+        .target = target,
+        .optimize = optimize
     });
+
+    lib.defineCMacro("_FILE_OFFSET_BITS", "64");
+    lib.defineCMacro("__STDC_CONSTANT_MACROS", null);
+    lib.defineCMacro("__STDC_FORMAT_MACROS", null);
+    lib.defineCMacro("__STDC_LIMIT_MACROS", null);
+    lib.linkSystemLibrary("z");
+    lib.linkLibC();
+    switch (target.result.os.tag) {
+        .linux => lib.linkSystemLibrary("LLVM-17"), // Ubuntu
+        .macos => {
+            lib.addLibraryPath(.{ .path = "/usr/local/opt/llvm/lib" });
+            lib.linkSystemLibrary("LLVM");
+        },
+        else => lib.linkSystemLibrary("LLVM"),
+    }
+
+    b.installArtifact(lib);
+
+    _ = try b.modules.put("llvm", &lib.root_module);
+
     _ = b.addModule("clang", .{
-        .source_file = .{
+        .root_source_file = .{
             .path = "src/clang.zig",
         },
     });
 
-    buildTests(b);
-
     const examples = b.option(bool, "Examples", "Build all examples [default: false]") orelse false;
-
     if (examples) {
-        buildExample(b, .{
+        buildExample(b, target, .{
             .filepath = "examples/sum_module.zig",
-            .target = target,
+            .target = target.query,
             .optimize = optimize,
         });
-        buildExample(b, .{
-            .filepath = "examples/fatorial_module.zig",
-            .target = target,
+        buildExample(b, target, .{
+            .filepath = "examples/factorial_module.zig",
+            .target = target.query,
             .optimize = optimize,
         });
     }
+
+    const tests = b.option(bool, "Tests", "Build all tests [default: false]") orelse false;
```

Comment:
Why add `-DTests`?
There's already a b.step called test.
https://github.com/kassane/llvm-zig/blob/54c645b6bc87c557430a01aad7e269c8efcf8ec8/build.zig#L126

**Current:** `zig build test`
**Changed:** `zig build -DTests test`

Reply:
It looked to me like the tests are being built even if you don't run the tests, so -DTests was added that the tests would not be built when using llvm-zig through the package manager. But I forgot about the workflow, which needs the test step being built so I'll revert it anyways.